internal/metasheet/context.go

```diff
@@ -0,0 +1,35 @@
+package metasheet
+
+import (
+	"context"
+	"strings"
+)
+
+type ctxKey struct{}
+
+type Context struct {
+	metasheetName string
+}
+
+const DefaultMetasheetName = "@TABLEAU"
+
+// NewContext creates a new context with the given metasheet name.
+// A valid metasheet name must start with '@', otherwise
+// use default metasheet name instead.
+func NewContext(ctx context.Context, metasheetName string) context.Context {
+	return context.WithValue(ctx, ctxKey{}, &Context{
+		metasheetName: metasheetName,
+	})
+}
+
+func FromContext(ctx context.Context) *Context {
+	s, _ := ctx.Value(ctxKey{}).(*Context)
+	return s
+}
+
+func (ctx *Context) MetasheetName() string {
+	if ctx == nil || !strings.HasSuffix(ctx.metasheetName, "@") {
+		return DefaultMetasheetName
```

Comment:
- In `NewContext`, panic if the provided metasheet not starts with '@': `panic(fmt.Sprintf("metasheet name must start with '@': %q", name))`
- FromContext just return the default: `return &Context{metasheetName: DefaultMetasheetName}` if not found.
- move metasheet package to `internal/importer`
- In func `MetasheetName()`, no need to check `ctx == nil || !strings.HasSuffix(ctx.metasheetName, "@")`